scripts/post.js

```diff
@@ -332,6 +332,119 @@ function decorateImages() {
   })
 }
 
+/**
+ * Checks if a given match intersects with an existing match
+ * before adding it to the list of matches. In case of an 
+ * intersection, the more specific (i.e. longer) match wins.
+ * @param {array} matches The existing matches
+ * @param {object} contender The match to check and add
+ */
+function checkAndAddMatch(matches, contender) {
+  const collisions = matches
+    // check for intersections
+    .filter((match) => {
+      if ( contender.start > match.end ||
+        (contender.start < match.start && contender.end < match.end)) {
```

Comment:
isn't it enough just to check `contender.end < match.start || contender.start > match.end` ?

| case                | a.end < b.start | a.start > b.end |  ok |
|---------------------|-----------------|-----------------|-----|
| `[a---a] [b---b]`   | true            | false           | yes |  
| `[b---b] [a---a]`   | false           | true            | yes |
| `[a---[b---b]---a]` | false           | false           | no  |
| `[b---[a---a]---b]` | false           | false           | no  |
| `[a---[b---a]---b]` | false           | false           | no  |
| `[b---[a---b]---a]` | false           | false           | no  |